ibuffer-project.el

```diff
@@ -63,18 +63,38 @@
 (require 'ibuffer)
 (require 'ibuf-ext)
 
+(defvar ibuffer-project-group-cache (make-hash-table)
+  "Cache for mappings from a directory to its group data.")
+
+(defun ibuffer-project-purge-group-cache ()
+  "Purge the group cache."
+  (setq ibuffer-project-group-cache (make-hash-table)))
+
 (defun ibuffer-project-root (buf)
   "Return a cons cell (project-root . root-type) for BUF."
   (unless (string-match-p "^ " (buffer-name buf))
-    (with-current-buffer buf
-      (let ((dir (cdr (project-current))))
-        (cond
-         (dir (cons dir 'project))
-         (default-directory (cons (abbreviate-file-name default-directory) 'directory)))))))
+    (let ((dir (buffer-local-value 'default-directory buf)))
+      (when dir
+        (or (gethash dir ibuffer-project-group-cache)
+            (let* ((remote (tramp-handle-file-remote-p dir))
```

Comment:
It mean that because of `(tramp-handle-file-remote-p dir)` all in files different folders via e.g. `sudo:` will appear under a single group, right?

Can you please make it customizable variable like `group-remote-by-host` with `nil` by default?

Thanks!

Reply:
I've added an option to specify a regular expression pattern.